presentation/backend/src/presentation-backend/NativePlatform.ts

```diff
@@ -180,8 +180,10 @@ export const createDefaultNativePlatform = (props: DefaultNativePlatformProps):
     public clearRulesets() {
       return this.handleVoidResult(this._nativeAddon.clearRulesets());
     }
-    public async handleRequest(db: any, options: string) {
-      const result = await this._nativeAddon.handleRequest(db, options);
+    public async handleRequest(db: any, options: string, cancelEvent?: BeEvent<() => void>) {
+      const response = this._nativeAddon.handleRequest(db, options);
+      cancelEvent?.addListener(() => { response.cancel(); });
```

Comment:
I think, `cancelEvent` will never be invoked more than once since I am deleting it after invocation in `PresentationRpcImpl._pendingRequests.cleanupHandler` function. Should I still remove the listener in case something changes in the future?

Reply:
I think it would be better to remove it if we don't want `cancel` to be invoked more than once. In your case it won't be invoked more than once but somebody else might use `PresentationManager` directly and raise `cancelEvent` multiple times.